freebites/product
# 47 Reporting and Sharing

# Description


Added functionality to the three dots in the top left corner of the screen on an individual post's page. When clicked a modal will appear on the bottom of the screen giving the options to report or share. On pressing share, a popup will appear providing different sharing options. On pressing report, the modal will change to have an empty text box in which the user can explain why they reported the post and then submit this report.

## Issues

Resolve #47 

## Screenshots


<img src="https://github.com/user-attachments/assets/623c370d-5832-4e7d-b0da-070343a5864d" width=150>
<img src="https://github.com/user-attachments/assets/b9c428cf-f64f-4710-a66e-6d37f9e975b2" width=150>
<img src="https://github.com/user-attachments/assets/281e7e38-87a2-4c51-9808-a68b6aaaad00" width=150>
<img src="https://github.com/user-attachments/assets/b0deab64-df4d-40e0-84cf-72d4369a64f0" width=150>
<img src="https://github.com/user-attachments/assets/aa921118-b3ad-4d7d-9a40-a0697a47dbcd" width=150>

## Test

- Select any post on the home page and open it
- Press the ... button in the top right corner.
- Try pressing the share button on the modal popup to send a (placeholder) link to someone
- Try pressing the report button to open the report modal
- Type something into the text box and see that the cancel button changes to a submit button
- Press submit and see the confirmation message in the modal
- Close the modal

## Possible Downsides

Currently we share are sharing a placeholder link when the user presses the share button, so this will need to be implemented later. The report submission also does not send the report message anywhere at this time, but it calls a function which will can be given that functionality.

## Additional Documentations

N/A
